docs/LoRA.md

```diff
@@ -120,73 +189,170 @@ if config.merged_hf_save_path is not None:
+for more details.
 
-You can change the LoRA parameters by adding a `lora` section, or providing appropriate command line flags:
+```bash
+virtualenv -p python3.8 helm
+source helm/bin/activate
+pip install crfm-helm
+```
 
-```yaml
-lora:
-  r: 16
-  dropout: 0.1
+The rest of this section follows the [HELM tutorial](https://crfm-helm.readthedocs.io/en/latest/tutorial/).
+
+### 2. Evaluate the model
+
+Run this command to evaluate the model on the GSM8K benchmark:
+
+```bash
+export MYMODEL="<somewhere>/gsm8k-llama2-merged"
+helm-run --run-specs gsm:model=$MYMODEL --enable-huggingface-models $MYMODEL --suite v1 --max-eval-instances 1000
 ```
 
-The default configs are available as [`alpaca-lora.yaml`](https://github.com/stanford-crfm/levanter/blob/main/examples/alpaca-lora/alpaca-lora.yaml)
-and [`alpaca-lora-llama2.yaml`](https://github.com/stanford-crfm/levanter/blob/main/examples/alpaca-lora/alpaca-lora-llama2.yaml)
+If you want to also evaluate the baseline Llama 2 model, you can do that by replacing `$MYMODEL` with
+`meta-llama/Llama-2-7b-hf`:
+
+```bash
+export HUGGING_FACE_HUB_TOKEN=${YOUR TOKEN HERE}
+export MYMODEL="meta-llama/Llama-2-7b-hf"
+helm-run --run-specs gsm:model=$MYMODEL --enable-huggingface-models $MYMODEL --suite v1 --max-eval-instances 1000
+```
 
+This will take a while.
 
-## Running Alpaca-Lora
+### 3. Summarize and view the results
 
-Running this modified script is basically identical to running the original script.
-For example:
+These commands will summarize the results and then start a webserver to view them:
 
 ```bash
-infra/babysit-tpu-vm.sh llama-32 -z us-east1-d -t v3-32 --preemptible -- \
-WANDB_API_KEY=${YOUR TOKEN HERE} \
-HUGGING_FACE_HUB_TOKEN=${YOUR TOKEN HERE} \
-bash levanter/infra/run.sh python \
-levanter/examples/alpaca-lora/alpaca_lora.py \
---config_path levanter/examples/alpaca-lora/alpaca-lora-llama2.yaml \
---trainer.checkpointer.base_path gs://<somewhere> \
---hf_save_path gs://<somewhere>
+helm-summarize --suite v1
+helm-server --suite v1
 ```
 
+Then, you can go to `http://0.0.0.0:8000/?group=gsm` to see the results. That will show a table of the results like this:
+
+![results on gsm8k. Our lora model gets 0.252 exact match to Llama 2's 0.133.](figures/helm-gsm8k-results.png)
+
+From the table, we can see that the LoRA model is much better than the baseline Llama 2 model, almost doubling the accuracy (from 0.133 to 0.252)!
+If you compare it to [the official HELM results](https://crfm.stanford.edu/helm/latest/#/groups/gsm), you'll see that
+it's a bit better than Llama 13B, which is pretty good for a first try!
+
+You can also look at the individual results with the HELM UI. If you click through to the individual results for a model,
+you can see the model's predictions and the reference answers:
+
+![Example question that the LoRA model happens to get right](figures/helm-instance-example.png)
 
-## Using the model
+## Using the Model
```

Comment:
Rather than the order of Training -> HELM eval -> Use the Model, I personally think it would be smoother if we write in the order of Run Training -> Use the Model -> HELM evaluation. What do you think?

Reply:
sure ok